### PR TITLE
[SOAR-12421] Name Validator

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  -  Validate plugin name

## PR Requirements

Developers, verify you have completed the following items by checking them off:

- [ ] Unit tests written for any new or updated code
- [ ] Version bumped within setup.py
- [ ] Changelog entry added

### CODE REVIEW

File: icon_validator/rules/plugin_validators/name_validator.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from icon_validator.exceptions import ValidationException
+
+class NameValidator(KomandPluginValidator):
+
+    @staticmethod
+    def validate_name(name, plugin_name=False):
+        if not isinstance(name, str):
+            raise ValidationException("Name must not be blank")
+        if name == "":
+            raise ValidationException("Name must not be blank")
```
Comment (igorski-r7) [inline icon_validator/rules/plugin_validators/name_validator.py:11]: We can connect that cases together 😄 
```suggestion
        if not isinstance(name, str) or name == "":
            raise ValidationException("Name must not be blank")
```

File: icon_validator/rules/plugin_validators/name_validator.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    @staticmethod
+    def validate_plugin_name(spec):
+        if "name" not in spec.spec_dictionary():
+            raise ValidationException("Plugin name is missing.")
+
+        try:
+            NameValidator.validate_name(spec.spec_dictionary()["name"], plugin_name=True)
+        except Exception as e:
+            raise ValidationException("Plugin name not valid.", e)
```
Comment (igorski-r7) [inline icon_validator/rules/plugin_validators/name_validator.py:29]: Single character variable 😄 
```suggestion
        except Exception as error:
            raise ValidationException("Plugin name not valid.", error)
```

File: unit_test/test_validate_plugin.py [hunk 3]
```diff
@@ -735,6 +736,54 @@ def test_supported_version_validator_sup_vers_empty(self):
         result = validate(directory_to_test, file_to_test, False, True, [SupportedVersionValidator()])
         self.assertEqual(result, 1)
 
+    def test_good_name_validator(self):
+        directory_to_test = "plugin_examples/name_tests"
+        file_to_test = "good_name.spec.yaml"
```
Comment (igorski-r7) [inline unit_test/test_validate_plugin.py:741]: Kindly please move that to const variables above 😄

File: unit_test/test_validate_plugin.py [hunk 4]
```diff
@@ -735,6 +736,54 @@ def test_supported_version_validator_sup_vers_empty(self):
...
+        directory_to_test = "plugin_examples/name_tests"
+        file_to_test = "uppercase_name.spec.yaml"
+        result = validate(directory_to_test, file_to_test, False, True, [NameValidator()])
+        self.assertEqual(result, 1)
+
+    def test_whitespace_name_validator(self):
+        directory_to_test = "plugin_examples/name_tests"
+        file_to_test = "whitespace_name.spec.yaml"
+        result = validate(directory_to_test, file_to_test, False, True, [NameValidator()])
+        self.assertEqual(result, 1)
```
Comment (igorski-r7) [inline unit_test/test_validate_plugin.py:785]: Could you please connect all the tests with `parameterized`?

**Review by igorski-r7 — CHANGES_REQUESTED** [review]

File: unit_test/test_validate_plugin.py (hunk 3 above)
Comment (ablakley-r7) [reply unit_test/test_validate_plugin.py]: Let me know your thoughts.

**Review by igorski-r7 — APPROVED** [review]

**Review by dlaverty-r7 — APPROVED** [review]

File: icon_validator/rules/plugin_validators/name_validator.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+        if len(name.split("_")) > 7:
+            raise ValidationException(f"Name is too long, 7 words or less: contains {str(len(name.split('_')))}")
+
+    @staticmethod
+    def validate_plugin_name(spec):
+        if "name" not in spec.spec_dictionary():
+            raise ValidationException("Plugin name is missing.")
+        try:
+            NameValidator.validate_name(spec.spec_dictionary()["name"], plugin_name=True)
+        except Exception as error:
```
Comment (cmcnally-r7) [inline icon_validator/rules/plugin_validators/name_validator.py:25]: Is it possible to specify the type of error here or is it better to handle all exceptions and assume they are related to plugin name not found

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

**Pull request merged**